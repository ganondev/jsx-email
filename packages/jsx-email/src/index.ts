export * from './components/body';
export * from './components/button/button';
export * from './components/code';
export * from './components/color-scheme';
export * from './components/column';
export * from './components/conditional';
export * from './components/container';
export * from './components/font';
export * from './components/head';
export * from './components/heading';
export * from './components/hr';
export * from './components/html';
export * from './components/img';
export * from './components/link';
export * from './components/markdown';
export * from './components/preview';
export * from './components/row';
export * from './components/section';
export * from './components/tailwind/tailwind';
export * from './components/text';
export * from './renderer/render';
export * from './renderer/compat/context';
export * from './renderer/compat/hooks';
export * from './renderer/jsx-to-string';
export { useData } from './renderer/suspense';
export * from './types';
