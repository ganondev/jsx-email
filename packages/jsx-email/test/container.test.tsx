// @ts-ignore
import React from 'react';

import { jsxToString } from '../src/renderer/jsx-to-string';
import { Container } from '../src';

describe('<Container> component', async () => {
  beforeEach(() => {
    vi.restoreAllMocks();
    vi.resetModules();
  });

  it('renders children correctly', async () => {
    const testMessage = 'Test message';
    const html = await jsxToString(<Container>{testMessage}</Container>);
    expect(html).toContain(testMessage);
  });

  it('passes style and other props correctly', async () => {
    const style = { backgroundColor: 'red', maxWidth: 300 };
    const html = await jsxToString(
      <Container style={style} data-testid="container-test">
        Test
      </Container>
    );
    expect(html).toContain('style="max-width:300px;background-color:red"');
    expect(html).toContain('data-testid="container-test"');
  });

  it('renders correctly', async () => {
    const container = await jsxToString(
      <Container style={{ maxWidth: '300px' }}>
        <button>Hi</button>
      </Container>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders alternate alignment', async () => {
    const container = await jsxToString(
      <Container alignment="right">
        <button>Hi</button>
      </Container>
    );

    expect(container).toMatchSnapshot();
  });
});
