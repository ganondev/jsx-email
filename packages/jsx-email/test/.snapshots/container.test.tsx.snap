// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Container> component > renders alternate alignment 1`] = `"<table align=\\"right\\" width=\\"100%\\" role=\\"presentation\\" cellSpacing=\\"0\\" cellPadding=\\"0\\" border=\\"0\\" style=\\"max-width:37.5em\\"><tbody><tr style=\\"width:100%\\"><td align=\\"right\\"><button>Hi</button></td></tr></tbody></table>"`;

exports[`<Container> component > renders correctly 1`] = `"<table align=\\"center\\" width=\\"100%\\" role=\\"presentation\\" cellSpacing=\\"0\\" cellPadding=\\"0\\" border=\\"0\\" style=\\"max-width:300px\\"><tbody><tr style=\\"width:100%\\"><td align=\\"center\\"><button>Hi</button></td></tr></tbody></table>"`;
